operator/src/lib.rs

```diff
@@ -0,0 +1,608 @@
+#![feature(backtrace)]
+mod config;
+mod error;
+
+use crate::error::Error;
+
+use kube::Api;
+use tracing::{debug, error, info, trace};
+
+use k8s_openapi::api::core::v1::{
+    ConfigMap, ConfigMapVolumeSource, Container, EnvVar, Pod, PodSpec, Volume, VolumeMount,
+};
+use kube::api::{ListParams, Meta, ObjectMeta};
+use serde_json::json;
+
+use async_trait::async_trait;
+use handlebars::Handlebars;
+use stackable_operator::client::Client;
+use stackable_operator::controller::{Controller, ControllerStrategy, ReconciliationState};
+use stackable_operator::metadata;
+use stackable_operator::reconcile::{
+    ReconcileFunctionAction, ReconcileResult, ReconciliationContext,
+};
+use stackable_operator::{create_config_map, create_tolerations};
+use stackable_spark_crd::{
+    SparkCluster, SparkClusterSpec, SparkClusterStatus, SparkNodeSelector, SparkNodeType,
+};
+use std::collections::{BTreeMap, HashMap};
+use std::future::Future;
+use std::pin::Pin;
+use std::str::FromStr;
+use uuid::Uuid;
+
+const FINALIZER_NAME: &str = "spark.stackable.de/cleanup";
+
+const HASH: &str = "spark.stackable.de/hash";
```

Comment:
Could you rename them to HASH_LABEL and TYPE_LABEL and maybe document what they mean?